lib/charms/traefik_k8s/v1/ingress.py

```diff
@@ -502,7 +538,12 @@ def provide_ingress_requirements(self, *, host: str = None, port: int):
     @property
     def relation(self):
         """The established Relation instance, or None."""
-        return self.relations[0] if self.relations else None
+        relations = self.relations
+        if not relations:
+            return None
+        if len(relations) > 2:
+            raise RuntimeError("Too many ingress relations: {}".format(relations))
```

Comment:
May as well include a suggestion for `limit: 1` in the message as a pointer to authors who may know that option in the metadata.